### DISCUSSION TITLE
Sub-menu question

### DISCUSSION THREAD

**Discussion by axtens** [unoplatform/uno#14015] [Q&A] [answered]
Regarding menus in Uno, I'm trying to use MenuFlyoutSubItem as per this fragment
```xml
<MenuBarItem Title="Interpreter" BorderBrush="LightGray"  BorderThickness="1" x:Name="InterpreterBar" >
    <MenuFlyoutItem x:Name="RunningMode" Text="Running mode" Click="InterpretBar_RunningMode_Click">
        <MenuFlyoutSubItem x:Name="RunningModeItems">
            <MenuFlyoutItem x:Name="RunningMode_Quiet" Text="Quiet" />
            <MenuFlyoutItem x:Name="RunningMode_Verbose" Text="Verbose" />
            <MenuFlyoutItem x:Name="RunningMode_VerbosePaus" Text="Verbose, with pause" />
        </MenuFlyoutSubItem>
    </MenuFlyoutItem>
```
but I get an error regarding too many Texts.

I have tried other arrangements to the same outcome.

How do I use MenuFlyoutSubItem? I want to be able to equivalence a radio-button approach where I have a tick-glyph marking the currently selected item. I understand that in the above XAML there needs to be a method called "InterpretBar_RunningMode_Click" that will handle the selection of the sub. If the selected item already has a tick-glyph, then it will be left there. If it does not have a tick-glyph, the glyph will be removed from any other it may be on and applied to the current. What make it more achievable is, on entry to the method, to strip all the ticks and then apply to the selected item even it it had one.

**Comment by jeromelaban** [comment]
Thanks for the question! 

It seems that what you're looking for is `ToggleMenuFlyoutItem`, you can see an example here: https://playground.platform.uno/#a75b7100. 

You'll like need to data bind menu items to something that will make only one menu item checked.

Could you also post the complete error you're getting? It sounds unfamiliar.

**Comment by axtens** [comment, marked as answer]
I've found a fix that's going to do the job. But the original posting stems from ignorance about what the Sub's syntax is.
```xml
<MenuFlyoutSubItem Text="Running Mode">
<MenuFlyoutItem Text="Quiet"/>
<MenuFlyoutItem Text="Verbose"/>
<MenuFlyoutItem Text="Verbose, with pause"/>
</MenuFlyoutSubItem>
```
It seems that you put MenuFlyoutItem inside MenuFlyoutSubItem. Not particularly intuitive.

**Reply by MartinZikmund** [reply]
Yes, that is the syntax used by WinUI - https://learn.microsoft.com/en-us/windows/apps/design/controls/menus and https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.menuflyoutsubitem
